packages/edit-widgets/src/store/resolvers.js

```diff
@@ -18,16 +18,11 @@ import {
 } from './utils';
 import { transformWidgetToBlock } from './transformers';
 
-export function* getWidgetAreas() {
+export const getWidgetAreas = () => async ( { dispatch, registry } ) => {
 	const query = buildWidgetAreasQuery();
-	yield resolveWidgetAreas( query );
-	const widgetAreas = yield select(
-		'core',
-		'getEntityRecords',
-		KIND,
-		WIDGET_AREA_ENTITY_TYPE,
-		query
-	);
+	const widgetAreas = await registry
+		.resolveSelect( 'core' )
```

Comment:
```suggestion
		.resolveSelect( coreStore )
```

It looks like the ESLint rule doesn't work in this scenario.